### PR TITLE
Add a QueryBuilder based Mapper

### PR DESCRIPTION
**Pull request by rullzer**
I noticed when I wanted to log the queries we make that not all of them are logged. Then I remembered that the Mapper doesn't use the query builder yet. So I decided to add a new one.

* Add new QBMapper
* Deprecate old Mapper
* Moved over DefaultTokenMapper

We should probably add tests.

### CODE REVIEW

**Review by ChristophWurst — APPROVED** [review]
:+1: looks great

**Review by MorrisJobke — APPROVED** [review]
Code makes sense 👍 - please add it to the docs.

File: lib/public/AppFramework/Db/QBMapper.php [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+		$qb = $this->db->getQueryBuilder();
+		$qb->insert($this->tableName);
+
+		// build the fields
+		foreach($properties as $property => $updated) {
+			$column = $entity->propertyToColumn($property);
+			$getter = 'get' . ucfirst($property);
+			$value = $entity->$getter();
+
+			$qb->setValue($column, $qb->createNamedParameter($value));
```
Comment (MorrisJobke) [inline lib/public/AppFramework/Db/QBMapper.php]: lib/public/AppFramework/Db/QBMapper.php:115 SqlInjectionChecker Potential SQL injection detected - method: no child method

File: lib/public/AppFramework/Db/QBMapper.php [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+		$qb = $this->db->getQueryBuilder();
+		$qb->update($this->tableName);
+
+		// build the fields
+		foreach($properties as $property => $updated) {
+			$column = $entity->propertyToColumn($property);
+			$getter = 'get' . ucfirst($property);
+			$value = $entity->$getter();
+
+			$qb->set($column, $qb->createNamedParameter($value));
```
Comment (MorrisJobke) [inline lib/public/AppFramework/Db/QBMapper.php]:
```
SqlInjectionChecker Potential SQL injection detected - method: no child method
```

**Pull request merged**